api-extractor/src/definitions/ApiItem.ts

```diff
@@ -247,6 +260,13 @@ abstract class ApiItem {
   }
 
   /**
+   * Called after the constructor to finish the analysis.
+   */
+  public resolveReferences(): void {
```

Comment:
Maybe we could improve these names:

- Visit the ApiItem tree:  resolveReferences()  -->  visitTypeReferencesForApiItem
- Visit a TypeScript expression:  collectTypeReferences()  -->  visitTypeReferencesForNode
- Analyze and report problems:  _processTypeReference()  -->  analyzeTypeReference